### ISSUE TITLE
upgrades to boosted_trees: an example with feature importance + tuning + savedmodel

### ISSUE THREAD

**Issue by evanfwelch** [tensorflow/tensorflow#14520]
### System information
N/A

### Describe the problem
*Feature Request*: I hate to make a trite request, but is there any plan to add more documentation or examples to the `tensorflow.contrib.boosted_trees` submodule? Now that there's a GB classifier available, I am hoping to replace my production XGBoost code with `boosted_trees` as soon as I can prove it out. Perhaps it's my newness to `tf.contrib.learn`'s `Estimator`s and `Experiment`s, but I found the examples a bit opaque and repetitive. For example, would it be possible to have an example script demonstrating these properties:
* some feature importance metrics (I see the Hooks but don't know how to use them)
* some type of hyperparameter tuning (max_depth, examples_per_layer, etc.)
* the implementation of an ExportStrategy in the Experiment; or better yet, throw away the experiment and directly operate on a trained Estimator. I'm struggling to write a proper serving input function
* a (perhaps commented out) example of using each input in the `GradientBoostedTreeClassifier` constructor

Sorry to ask for hand-holding here. It's just taking me longer than usual to figure these things out from the source code.

Also, to the extent that I've figured out some of these, are you welcoming PRs for more thorough examples?

**Comment by evanfwelch** [comment]
I also notice that `boosted_trees` is using what I think is a deprecated version of the `Estimator` class. I discovered this when `<instance of GBTClassifier>.train(input_fn=my_input_fun)` throw an error. Seems like `.fit` accomplishes the same goal but this created some uncertainty about how compatible the `boosted_trees` estimators were with other `tf` features (like SavedModel).

**Comment by bignamehyp** [comment]
@ThomasColthurst

**Comment by sshrdp** [comment]
There are some examples [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/boosted_trees/examples/boston.py) but I agree that we should add the documentation and examples for the scenarios you mentioned (feature importance, exporting and predicting, etc). SavedModel export should work. 
boosted_trees estimator is implementing the older version of estimator (tf.contrib.learn.Estimator) that didn't have a train() but switching to the new Estimator is on the roadmap is well.

**Comment by evanfwelch** [comment]
Sounds great, thanks so much for addressing my questions.

As for generating a more thorough example, I'm happy to assist if you guys need help. If you DM me a snippet for feature importance I can probably figure out the rest.

**Comment by sshrdp** [comment]
I added an example [here](7c7ccb0ba476d12814b5be2a0b87f30784977a7e).  Contributions are also welcomed :) 
The serving_input_fn in the example takes in tensorflow.Example but you can see the implementation and adjust it based on the type of input you want to serve.

**Comment by evanfwelch** [comment]
Mind reposting the link to your example? I’m getting a 404 not found when I click on it. Thanks!

**Comment by sshrdp** [comment]
Here is the commit: https://github.com/tensorflow/tensorflow/commit/20e2fdc2f95f213eef5a736a140d8591ef7a5b6e

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.
